src/main/kotlin/com/vanniktech/maven/publish/MavenPublishPluginExtension.kt

```diff
@@ -7,31 +7,52 @@ import org.gradle.api.Project
  * @since 0.1.0
  */
 open class MavenPublishPluginExtension(project: Project) {
+
+  @JvmField
+  internal val defaultTarget = MavenPublishTarget(
```

Comment:
My idea for this is to add the Map that is in MavenPublishRepositories to this extension and put `defaultTarget` and `localTarget` in there by default. Afterwards the plugin can do the configuration completely based on the Map.